### PR TITLE
chore: upgrade cypress to 5.0.0

### PR DESCRIPTION
**Pull request by mroz22**
- upgrade cypress by one major version
- fix bug with running out of ip addresses in CI
- use global beforeEach to stop bridge and emulator before every test

### CODE REVIEW

**Review by keraf — APPROVED** [review]

File: ci/packages/suite-web.yml [hunk 1]
```diff
@@ -76,10 +76,16 @@ suite-web deploy dev:
...
+    # corresponds with the NextJs assetPrefix. Tells Cypress where is the index of application
     CYPRESS_ASSET_PREFIX: /web
     CYPRESS_baseUrl: ${DEV_SERVER_URL}/suite-web/${CI_BUILD_REF_NAME}
+    # should tests do snapshot testing
     CYPRESS_SNAPSHOT: 0
+    # reporter url
     TRACK_SUITE_URL: https://track-suite.herokuapp.com
+    # when debugging or developing tests it does not make sense to have retries,
+    # in other cases retries are useful to avoid occasional failures due to flaky tests
+    ALLOW_RETRY: 1
```
Comment (tsusanka) [inline ci/packages/suite-web.yml:88]: Btw gitlab can also do that which might be better to see which tests are flaky? But JFYI, no need to change anything.

File: docker/docker-compose.suite-ci.yml [hunk 2]
```diff
@@ -5,9 +5,10 @@ services:
     environment:
       - SDL_VIDEODRIVER=dummy
       - XDG_RUNTIME_DIR=/var/tmp
+    network_mode: bridge # this makes docker reuse existing networks
 
   test-run:
-    image: cypress/included:4.7.0
+    image: cypress/included:5.0.0
```
Comment (tsusanka) [inline docker/docker-compose.suite-ci.yml:11]: I have noticed that latest cypress is 6.x. Do we upgrade to 5.x just to have it gradually or we need to stick with 5 for some reason?

**Review by tsusanka — APPROVED** [review]
Can't really review the code changes since I do not know Cypress enough but I've left a few remarks on the CI part. Other than that LGTM.

File: docker/docker-compose.suite-ci.yml (hunk 2 above)
Comment (mroz22) [reply docker/docker-compose.suite-ci.yml:11]: Just to make this PR easier. 6.1.0 comes in another one https://github.com/trezor/trezor-suite/pull/3159

File: ci/packages/suite-web.yml (hunk 1 above)
Comment (mroz22) [reply ci/packages/suite-web.yml:88]: this is custom retry functionality controlled by run_tests.js file. Gitlab can rerun the entire job but this is suboptimal when I need to rerun only a single test.

File: docker/docker-compose.suite-ci.yml (hunk 2 above)
Comment (tsusanka) [reply docker/docker-compose.suite-ci.yml:11]: You are the best :rocket:

**Comment by mroz22** [comment]
it has 2 approvals so I am going to merge it myself.

**Pull request merged**